test/e2e/helpers/broker_with_config_helper.go

```diff
@@ -0,0 +1,154 @@
+		loggerPodName2        = "e2e-brokerchannel-logger-pod2"
+
+		channelName      = "e2e-brokerchannel-channel"
+		subscriptionName = "e2e-brokerchannel-subscription"
+	)
+
+	channelTestRunner.RunTests(t, lib.FeatureBasic, func(st *testing.T, channel metav1.TypeMeta) {
+		client := lib.Setup(st, true, options...)
+		defer lib.TearDown(client)
+
+		// create required RBAC resources including ServiceAccounts and ClusterRoleBindings for Brokers
+		client.CreateRBACResourcesForBrokers()
+
+		config := client.CreateBrokerConfigMapOrFail(brokerName, &channel)
+		//&channel
+
+		// create a new broker
+		client.CreateBrokerV1Beta1OrFail(brokerName, resources.WithChannelTemplateForBrokerV1Beta1(config))
+		client.WaitForResourceReadyOrFail(brokerName, lib.BrokerTypeMeta)
+
+		// create the event we want to transform to
+		transformedEventBody := fmt.Sprintf("%s %s", eventBody, string(uuid.NewUUID()))
+		eventAfterTransformation := cloudevents.New(
+			fmt.Sprintf(`{"msg":%q}`, transformedEventBody),
+			cloudevents.WithSource(eventSource2),
+			cloudevents.WithType(eventType2),
+		)
+
+		// create the transformation service for trigger1
+		transformationPod := resources.EventTransformationPod(transformationPodName, eventAfterTransformation)
+		client.CreatePodOrFail(transformationPod, lib.WithService(transformationPodName))
+
+		// create trigger1 to receive the original event, and do event transformation
+		client.CreateTriggerOrFail(
+			triggerName1,
+			resources.WithBroker(brokerName),
+			resources.WithDeprecatedSourceAndTypeTriggerFilter(eventSource1, eventType1),
+			resources.WithSubscriberServiceRefForTrigger(transformationPodName),
+		)
+
+		// create logger pod and service for trigger2
+		loggerPod1 := resources.EventLoggerPod(loggerPodName1)
+		client.CreatePodOrFail(loggerPod1, lib.WithService(loggerPodName1))
+
+		// create trigger2 to receive all the events
+		client.CreateTriggerOrFail(
+			triggerName2,
+			resources.WithBroker(brokerName),
+			resources.WithDeprecatedSourceAndTypeTriggerFilter(any, any),
+			resources.WithSubscriberServiceRefForTrigger(loggerPodName1),
+		)
+
+		// create channel for trigger3
+		client.CreateChannelOrFail(channelName, &channel)
+		client.WaitForResourceReadyOrFail(channelName, &channel)
+
+		// create trigger3 to receive the transformed event, and send it to the channel
+		channelURL, err := client.GetAddressableURI(channelName, &channel)
+		if err != nil {
+			st.Fatalf("Failed to get the url for the channel %q: %v", channelName, err)
+		}
+		client.CreateTriggerOrFail(
+			triggerName3,
+			resources.WithBroker(brokerName),
+			resources.WithDeprecatedSourceAndTypeTriggerFilter(eventSource2, eventType2),
+			resources.WithSubscriberURIForTrigger(channelURL),
+		)
+
+		// create logger pod and service for subscription
+		loggerPod2 := resources.EventLoggerPod(loggerPodName2)
+		client.CreatePodOrFail(loggerPod2, lib.WithService(loggerPodName2))
+
+		// create subscription
```

Comment:
Just having a little hard time decoding what all this test does, not sure if it mirrors an existing test, but just curious on why the need for subscriptions and channels if we're just trying to test the Broker with config :)

Reply:
I copied this from an original broker test and I also could not follow it. Our e2e tests are hard to read and write.